src/main/kotlin/frc/robot/subsystems/hood/HoodConstants.kt

```diff
@@ -1,80 +1,43 @@
 
-    val MOMENT_OF_INERTIA : Measure<Mult<Mult<Mass, Distance>, Distance>> = Units.Kilograms.mult(Units.Meters).mult(Units.Meters).of(0.0003);
+    val MOMENT_OF_INERTIA: Measure<Mult<Mult<Mass, Distance>, Distance>> =
+        Units.Kilograms.mult(Units.Meters).mult(Units.Meters).of(0.0003);
     val MAX_VELOCITY: Measure<Velocity<Angle>> = Units.RotationsPerSecond.of(1.0)
     val MAX_ACCELERATION: Measure<Velocity<Velocity<Angle>>> = Units.RotationsPerSecond.per(Units.Second).of(4.0)
     val RESTING_ANGLE: Measure<Angle> = Units.Degrees.of(100.0)
     val HOOD_LENGTH: Measure<Distance> = Units.Meters.of(0.4)
-   
+
     val MECHANISM_2D_POSE = Translation2d(1.0, 1.0)
     val ROOT_POSITION = Translation3d(-0.27, 0.0, 0.225)
     val SIMULATION_OFFSET: Measure<Angle> = Units.Degrees.of(-54.0)
 
-    private const val CURRENT_LIMIT = 40.0
-    private val INVERTED_VALUE = InvertedValue.CounterClockwise_Positive
-
+    const val CURRENT_LIMIT = 40.0
+    val INVERTED_VALUE = InvertedValue.CounterClockwise_Positive
 
-    val ABSOLUTE_ENCODER_OFFSET = LoggedTunableNumber("Hood/EncoderOffset")
 
-    val kP = LoggedTunableNumber("Hood/kP")
-    val kI = LoggedTunableNumber("Hood/kI")
-    val kD = LoggedTunableNumber("Hood/kD")
-    val kS = LoggedTunableNumber("Hood/kS")
-    val kV = LoggedTunableNumber("Hood/kV")
-    val kA = LoggedTunableNumber("Hood/kA")
-    val kG = LoggedTunableNumber("Hood/kG")
+    val ABSOLUTE_ENCODER_OFFSET = LoggedTunableNumber("Hood/EncoderOffset", (58.095 - 33.48) / 360.0)
 
-    val MOTOR_CONFIGURATION = TalonFXConfiguration()
-
-    init {
-        kP.initDefault(2400.0)
-        kI.initDefault(0.0)
-        kD.initDefault(600.0)
-        kS.initDefault(0.0)
-        kV.initDefault(0.0)
-        kA.initDefault(0.0)
-        kG.initDefault(9.0)
-        ABSOLUTE_ENCODER_OFFSET.initDefault((58.095 - 33.48) / 360.0)
-        MOTOR_CONFIGURATION
-            .withFeedback(FeedbackConfigs().withSensorToMechanismRatio(GEAR_RATIO))
-            .withMotionMagic(
-                MotionMagicConfigs()
-                    .withMotionMagicCruiseVelocity(
-                        MAX_VELOCITY.`in`(Units.RotationsPerSecond)
-                    )
-                    .withMotionMagicJerk(16.0)
-            )
-            .withSlot0(
-                Slot0Configs()
-                    .withKP(kP.get())
-                    .withKI(kI.get())
-                    .withKD(kD.get())
-                    .withKS(kS.get())
-                    .withKV(kV.get())
-                    .withKA(kA.get())
-                    .withKG(kG.get())
-                    .withGravityType(GravityTypeValue.Arm_Cosine)
-                    .withStaticFeedforwardSign(StaticFeedforwardSignValue.UseClosedLoopSign)
+    val GAINS by lazy {
+        selectGainsBasedOnMode(
+            Gains(
+                2400.0, 0.0, 600.0, 0.0, 0.0, 0.0, 9.0
+            ), Gains(
+                20.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0
```

Comment:
`.0` is redundant

Reply:
In kotlin you have to add .0 to doubles, otherwise they'd be integers.